src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -0,0 +1,118 @@
+        /// <param name="cancellationTokenSource">The <see cref="CancellationTokenSource"/> that can be used to cancel this <c>async</c> operation.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpMessageHandler")]
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpClientHandler")]
+        public static object SendAsync(
+            object handler,
+            object request,
+            object cancellationTokenSource)
+        {
+            // HttpMessageHandler
+            // Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            var tokenSource = cancellationTokenSource as CancellationTokenSource;
+            var cancellationToken = tokenSource?.Token ?? CancellationToken.None;
+
+            return SendAsyncInternal(
+                (HttpMessageHandler)handler,
+                (HttpRequestMessage)request,
+                cancellationToken);
+        }
+
+        private static async Task<HttpResponseMessage> SendAsyncInternal(
+            HttpMessageHandler handler,
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            var executeAsync = DynamicMethodBuilder<Func<HttpMessageHandler, HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>
+               .GetOrCreateMethodCallDelegate(
+                    handler.GetType(),
+                    nameof(SendAsync));
+
+            if (!IsTracingEnabled(request))
+            {
+                return await executeAsync(handler, request, cancellationToken).ConfigureAwait(false);
+            }
+
+            using (var scope = CreateScope(handler, request))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    request.Headers.Inject(scope.Span.Context);
+
+                    HttpResponseMessage response = await executeAsync(handler, request, cancellationToken).ConfigureAwait(false);
+                    scope.Span.SetTag(Tags.HttpStatusCode, ((int)response.StatusCode).ToString());
+                    return response;
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        private static bool IsTracingEnabled(HttpRequestMessage request)
+        {
+            if (request.Headers.TryGetValues(HttpHeaderNames.TracingEnabled, out var headerValues))
+            {
+                if (headerValues.Any(s => string.Equals(s, "false", StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Changed to `OrdinalIgnoreCase` in 5e54c75bb43092b1ce66271458dcdd4848239745.

I'm not going to bother with an extension method that turns
```csharp
s.Equals("false", StringComparison.OrdinalIgnoreCase)
```
into 
```csharp
s.EqualsOrdinalIgnoreCase("false")
```

The reason I picked the static `string.Equals()` instead of the instance method is because it automatically takes care of null values.